HW5(OOP_vs_FP)/HW5_Trupkina.js

```diff
@@ -0,0 +1,236 @@
+//(4) [23, 36, 20, 15]
+
+//1.2 hof
+//create array fhat contain current age in 2021 with hof
+function findAgeIn2021_hof(arr, fn) {
+    const arr2021 = [];
+    for (let i = 0; i < arr.length; i++) {
+        arr2021.push(fn(arr[i]));//call special func for calculate age 
+    }
+    return arr2021;
+}
+function findNowAge(elem) {
+    const ageNow = 2021 - elem;
+    return ageNow;
+}
+console.log(findAgeIn2021_hof(arrBirthday, findNowAge));
+//(4) [23, 36, 20, 15]
+
+//1.2.1 hof map
+const arrAgeNow_hof = arrBirthday.map(function (elem) {
+    return elem = (2021 - elem);
+});
+console.log(arrAgeNow);
+//(4) [23, 36, 20, 15]
+
+// 2.
+//2.1
+const arrPeople = [
+    { name: 'Pedro Pascal', age: 45 },
+    { name: 'Leonardo Dicaprio', age: 46 },
+    { name: 'Thomas A. Anderson', age: 58 },
+    { name: 'Millie Brown', age: 17 },
+    { name: 'Harry Potter', age: 31 }
+];
+function findAdults(arr) {
+    const arrOver18 = [];
+    for (let i = 0; i < arr.length; i++) {
+        if (arr[i].age >= 18) {
+            arrOver18.push(arr[i]);
+        }
+    }
+    return arrOver18;
+}
+console.log(findAdults(arrPeople));
+/*(4) [{…}, {…}, {…}, {…}]
+0: {name: "Pedro Pascal", age: 45}
+1: {name: "Leonardo Dicaprio", age: 46}
+2: {name: "Thomas A. Anderson", age: 58}
+3: {name: "Harry Potter", age: 31} */
+
+//2.2 hof
+function findAdults_hof(arr, fn) {
+    const arrOver18 = [];
+    for (let i = 0; i < arr.length; i++) {
+        if (fn(arr[i]) === undefined) i++;
+        arrOver18.push(fn(arr[i]));
+    }
+    return arrOver18;
+}
+function more18(elem) {
+    if (elem.age >= 18) return elem.name;
+}
+console.log(findAdults_hof(arrPeople, more18));
+//(4) ["Pedro Pascal", "Leonardo Dicaprio", "Thomas A. Anderson", "Harry Potter"]
+
+//2.2.1 hof filter
+const findAdults_filter = arrPeople.filter(arrPeople => arrPeople.age >= 18);
+console.log(findAdults_filter);
+/*(4) [{…}, {…}, {…}, {…}]
+0: {name: "Pedro Pascal", age: 45}
+1: {name: "Leonardo Dicaprio", age: 46}
+2: {name: "Thomas A. Anderson", age: 58}
+3: {name: "Harry Potter", age: 31}*/
+
+
+// 3.
+//3.1
+const arrNum = [34, 38, 45, 9, 23];
+function findSumArrNumb(numb) {
+    let sum = 0;
+    for (let i = 0; i < numb.length; i++) {
+        sum += numb[i];
+    }
+    return sum;
+}
+console.log(findSumArrNumb(arrNum));//149
+
+
+//3.2.1 reduce________________________________
+const sum = arrNum.reduce(function (accumulator, currentValue) { //.reduce(function(accum, currentValue, index[0...n], array) {
+    return accumulator + currentValue;
+});
+console.log(sum);//25
+
+//   4.
+const arrPeopleName = ['Pedro', 'Leonardo', 'Anderson', 'Millie', 'Harry'];
+
+function findNameLength_hof(arr, fn) {
+    const arrlen = [];
+    for (let i = 0; i < arr.length; i++) {
+        arrlen.push(fn(arr[i]));
+    }
+    return arrlen;
+}
+function retName(item) {
+    return item.length;
+}
+const arrMap = findNameLength_hof(arrPeopleName, retName);
+console.log(arrMap);
```

Comment:
pls, try emulate "map" method. in the end we need something like 

arrayName.newMap(someFunc);